addon-manager/impl/src/main/java/org/jboss/forge/addon/manager/impl/ui/DistributionUpdateCommand.java

```diff
@@ -0,0 +1,161 @@
+       File addonsDir = new File(forgeHome.getAbsolutePath() + "/addons");
+       File binDir = new File(forgeHome.getAbsolutePath() + "/bin");
+       File libDir = new File(forgeHome.getAbsolutePath() + "/lib");
+
+       deleteWholeDirectory(addonsDir);
+       deleteWholeDirectory(libDir);
+       deleteWholeDirectory(binDir);
+       Dependency dependency = dependencyResolver.resolveArtifact(DependencyQueryBuilder.create(CoordinateBuilder.create()
+                   .setGroupId(latestDistributionCoordinate.getGroupId()).setArtifactId(latestDistributionCoordinate.getArtifactId())
+                   .setVersion(latestDistributionCoordinate.getVersion()).setClassifier("offline").setPackaging("zip")));
+       FileResource<?> artifact = dependency.getArtifact();
+       ZipFile zipFile = new ZipFile(artifact.getFullyQualifiedName());
+       File tempFolder = OperatingSystemUtils.createTempDir();
+       String tempFolderPath = tempFolder.getAbsolutePath();
+       zipFile.extractAll(tempFolderPath);
+       FilenameFilter windupUnzipped = new FilenameFilter()
+       {
+           public boolean accept(File dir, String name)
+           {
+               String lowercaseName = name.toLowerCase();
+               if (lowercaseName.contains("forge-distribution"))
+               {
+                   return true;
+               }
+               else
+               {
+                   return false;
+               }
+           }
+       };
+       File[] windupDistrubution = new File(tempFolderPath).listFiles(windupUnzipped);
+       String distributionExctractedPath = windupDistrubution[0].getAbsolutePath();
+       FileUtils.copyDirectory(new File(distributionExctractedPath + "/" + "addons"), addonsDir);
+       FileUtils.copyDirectory(new File(distributionExctractedPath + "/" + "bin"), binDir);
+       FileUtils.copyDirectory(new File(distributionExctractedPath + "/" + "lib"), libDir);
+       return Results.success("Sucessfully updated to version " + latestDistributionCoordinate.getVersion() + ". Please restart windup.");
+   }
+
+   private void deleteWholeDirectory(File directory) throws IOException
+   {
+       if (directory.exists())
+       {
+           Files.walkFileTree(directory.toPath(), new SimpleFileVisitor<Path>()
+           {
+               @Override
+               public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                           throws IOException
+               {
+                   Files.delete(file);
+                   return FileVisitResult.CONTINUE;
+               }
+
+               @Override
+               public FileVisitResult postVisitDirectory(Path dir,
+                           IOException exc) throws IOException
+               {
+                   Files.delete(dir);
+                   return FileVisitResult.CONTINUE;
+               }
+           });
+       }
+   }
+
+   @Override
+   public boolean isEnabled(UIContext context)
+   {
+       return true;
```

Comment:
Oh, I finally get what's this enabled about (always seen just returning true) :)